src/zodotenv.ts

```diff
@@ -35,7 +35,7 @@ const walk = (map: Map<string, unknown>, entry: ZodotenvConfig | EnvWithZodType,
       );
     }
 
-    map.set(prefix, data);
+    map.set(prefix, { value: data, secret: options?.secret });
```

Comment:
shall we store the entire options instead?
```suggestion
    map.set(prefix, { value: data, options });
```